src/main/java/models/StringData.java

```diff
@@ -0,0 +1,31 @@
+package models;
+
+import java.util.Optional;
+
+public class StringData {
```

Comment:
Generally, javadocs on interfaces and DA for sure, maybe utility methods. Everything else should just be fairly simple, but more javadocs is better.

As far as unit tests, we should have "good enough" `JUnit` tests for everything. Obviously we don't want to merge something if it's big enough that it might not work and doesn't have tests.